staging/src/k8s.io/apimachinery/pkg/util/httpstream/wsstream/conn.go

```diff
@@ -87,6 +89,19 @@ func IsWebSocketRequest(req *http.Request) bool {
 	return httpstream.IsUpgradeRequest(req)
 }
 
+// IsWebSocketRequestedProtocol returns true if the passed streaming protocol is one
+// of the requested protocols for websocket upgrade in the passed http request; false otherwise.
+func IsWebSocketRequestedProtocol(req *http.Request, protocol string) bool {
+	protocol = strings.TrimSpace(strings.ToLower(protocol))
```

Comment:
this doesn't look right, in a few ways:
1. protocol value is case-sensitive
2. the header can specify multiple protocols, comma-separated
3. this should be consistent with how the server detects / negotiates the protocol, and it looks like we're currently only paying attention to the first header value, not honoring repeated headers:

https://github.com/kubernetes/kubernetes/blob/1a1a3d0a87530aea0294ce7ce91a79e7378b95a7/vendor/golang.org/x/net/websocket/hybi.go#L523-L529

https://github.com/kubernetes/kubernetes/blob/1a1a3d0a87530aea0294ce7ce91a79e7378b95a7/vendor/github.com/gorilla/websocket/server.go#L307-L319

Reply:
make sure there's a test that would have caught the case-sensitive and multi-protocol requests

Reply:
see the suggestion below about a `IsWebSocketRequestWithStreamCloseProtocol` method instead of this

Reply:
I believe this is addressed.